crates/common/mqtt_channel/src/connection.rs

```diff
@@ -166,6 +170,27 @@ impl Connection {
                     let _ = message_sender.send(msg.into()).await;
                 }
 
+                Ok(Event::Incoming(Packet::ConnAck(ack))) => {
+                    // If session_name is not provided, then re-subscribe
```

Comment:
I would mark this as a workaround

```suggestion
                    // Workaround for  https://github.com/bytebeamio/rumqtt/issues/250
                    // If session_name is not provided, then re-subscribe
```